docs/user-guide/user-defined-functions/map/map.md

```diff
@@ -51,6 +51,29 @@ Check the links below to see the UDF examples in streaming mode for different la
 - [Golang](https://github.com/numaproj/numaflow-go/tree/main/pkg/mapstreamer/examples/flatmap_stream/)
 - [Java](https://github.com/numaproj/numaflow-java/tree/main/examples/src/main/java/io/numaproj/numaflow/examples/mapstream/flatmapstream/)
 
+### Batch Map Mode
+
+BatchMap is an interface that allows developers to process multiple data items in a UDF single call,
+rather than each item in separate calls.
+
+The BatchMap interface can be helpful in scenarios where performing operations on a group of data can be more efficient.
```

Comment:
please add that the batch size might be smaller (as small as one element) and the size is best effort..